src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+
+public class FruitDaoImpl implements FruitDao {
+    private static final String MESSAGE_FRUIT_NOT_AVAILABLE = "Sorry, but we don't have ";
+    private static final String MESSAGE_QUANTITY_NOT_ENOUGH = "We don't have enough quantity ";
+    private final Storage storage;
+
+    public FruitDaoImpl(Storage storage) {
+        this.storage = storage;
+    }
+
+    public void supplyFruit(String fruit, int quantity) {
+        var balanceOfQuantity = storage.getStorage().get(fruit);
+        if (storage.getStorage().containsKey(fruit)) {
+            storage.getStorage().replace(fruit, balanceOfQuantity + quantity);
+        } else {
+            storage.getStorage().put(fruit, quantity);
+        }
+    }
+
+    @Override
+    public void boughtFruit(String fruit, int quantity) {
+        var balanceOfQuantity = storage.getStorage().get(fruit);
+        if (!storage.getStorage().containsKey(fruit)) {
+            System.out.println(MESSAGE_FRUIT_NOT_AVAILABLE + fruit);
+            return;
+        } else if (storage.getStorage().get(fruit) >= quantity) {
+            storage.getStorage().replace(fruit, balanceOfQuantity - quantity);
+            return;
+        }
+        System.out.println(MESSAGE_QUANTITY_NOT_ENOUGH + fruit);
```

Comment:
In the `boughtFruit` method, you should check if the resulting balance is negative before updating the storage. If the balance is negative, throw a `RuntimeException` with an informative message. This aligns with the checklist requirement: "Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative."